src/main/java/seedu/address/model/person/LicenceContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Licence Plate} matches any of the keywords given.
+ */
+public class LicenceContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public LicenceContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getLicencePlate().value, keyword));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof LicenceContainsKeywordsPredicate)) {
+            return false;
+        }
+
+        LicenceContainsKeywordsPredicate otherNameContainsKeywordsPredicate = (LicenceContainsKeywordsPredicate) other;
+        return keywords.equals(otherNameContainsKeywordsPredicate.keywords);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this).add("keywords", keywords).toString();
+    }
+
+    public boolean isEmpty() {
+        return keywords.contains("");
```

Comment:
Should this be `equals("")` instead of `contains("")`?

Reply:
Because keywords is a List<String> and not just a String hence .equals("") would not work here.